accepted/2024/package-update-command.md

```diff
@@ -0,0 +1,238 @@
+
+### Existing capabilities
+
+Visual Studio
+
+- Has a means to show outdated packages
+- Has a means to show deprecate/vulnerable packages
+- Has a means to update all packages to latest
+- Has a means to update a single package to latest (which can be challenging)
+- Has a vulnerability filter, which in turn can allow you to update all vulnerable packages to latest
+dotnet
+
+CLI
+
+- Has a means to show outdated packages
+- Has a means to show deprecated/vulnerable packages
+- Has a means to update a single package
+- Does not have means to update multiple packages at once
+
+### Gaps in CLI experience
+
+- No means to update to all latest packages on the CLI
+- No means for update heuristics, such as, update my vulnerable packages
+
+This proposal is focused on bringing the CLI experience up to funcional parity with the experience in Visual Studio and make developers more productive by providing a single command to update all packages to the latest version and a way to update all vulnerable packages.
+
+## User experience
+
+### Updating all packages to the latest version
+
+When users want to ensure their NuGet dependencies are up to date, they can run a command in .NET CLI `dotnet package update` for a project or a solution.
+
+```CLI
+C:\> dotnet package update [<SOLUTION_PATH>|<PROJECT_PATH_>]
+
+Fixing outdated packages in ContosoUniversity.sln
+ ContosoLibrary:
+  Updated UmbracoForms 8.4.1 to 8.7.1
+  Updated Newtonsoft.Json 11.0.1 to 13.0.3
+
+ ContosoApp:
+  Updated UmbracoForms 8.4.0 to 8.7.1
+    
+Updated 3 packages in 36 scanned packages.
+```
+
+### Fixing vulnerabilities
+
+When users run `dotnet build` or `dotnet restore` commands in CLI, if they see any warnings related to vulnerabilities in their project’s NuGet dependencies, they can run `dotnet package update --vulnerable` CLI command to try to remediate all the vulnerabilities. This includes both direct and transitive.
+
+```CLI
+C:\ContosoApp\> dotnet package update --vulnerable
+
+Fixing vulnernable packages in ContosoUniversity.sln
+ ContosoLibrary:
+  Upgrading UmbracoForms 8.4.1 to 8.7.1
+
+ ContosoApp:
+  Updated UmbracoForms 8.4.0 to 8.7.1
+    
+Fixed 2 packages in 36 scanned packages.
+```
+
+In the future this can be extended to fix deprecated versions.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+
+#### dotnet package update
+
+The `dotnet package update` command will update all the NuGet dependencies to the latest versions.
+
+```CLI
+C:\> dotnet package update --help
+dotnet package update [<PROJECT>|<SOLUTION>] [--vulnerable] [--mode <MODE>] [-v|--verbosity <LEVEL>] [--json] 
```

Comment:
above, the json option was `--format json`, but here it's just `--json`.  As someone else commented, we already have prior art with `--format json`